doctr/transforms/modules/pytorch.py

```diff
@@ -119,3 +121,93 @@ def forward(
             _target["boxes"][:, ::2] = 1 - target["boxes"][:, [2, 0]]
             return _img, _target
         return img, target
+
+
+class RandomPerspective(T.RandomPerspective):
+
+    def forward(
+            self, img: Union[torch.Tensor, Image],
+            target: Dict[str, Any]
+    ) -> Tuple[Union[torch.Tensor, Image], Dict[str, Any]]:
+        """
+          Args:
+            img: Image to be transformed.
+            target: Dictionary with boxes (in relative coordinates of shape (N, 4)) and labels as keys
+        Returns:
+            Tuple of PIL Image or Tensor and target
+         """
+
+        fill = self.fill
+        if isinstance(img, torch.Tensor):
+            if isinstance(fill, (int, float)):
+                fill = [float(fill)] * self._get_num_channels(img)
+            else:
+                fill = [float(f) for f in fill]
+
+        if torch.rand(1) < self.p:
+            cmap = [(0, 0, 0), (230, 230, 230), (195, 195, 195), (175, 175, 175), (250, 250, 250)]
+            bbox = []
+            label = []
+
+            #  Preparing mask
+            if isinstance(img, torch.Tensor):
+                height, width = img.shape[-2:]
+                mask = np.zeros_like(img.permute(1, 2, 0))
+            else:
+                width, height = img.size
+                mask = np.zeros_like(img)
+
+            #  Drawing bounding boxes on mask with respective color
+            for ind, val in enumerate(target["boxes"]):
+                cv2.rectangle(
+                    mask, (int(val[0] * width), int(val[1] * height)),
+                    (int(val[2] * width), int(val[3] * height)),
+                    cmap[target["labels"][ind]], -1
+                )
+            mask = torch.from_numpy(mask).permute(2, 0, 1)
+            ddic = {}
+            startpoints, endpoints = self.get_params(width, height, self.distortion_scale)
+
+            #  transformed mask
+            transformed_mask = F.perspective(
+                mask, startpoints, endpoints, self.interpolation, fill
+            ).permute(1, 2, 0).numpy()
+
+            new_img = F.perspective(img, startpoints, endpoints, self.interpolation, fill)
+
+            #  filtering to separate bboxes of each label
+            mask_qr = cv2.inRange(transformed_mask, np.array([220, 220, 220]), np.array([240, 240, 240]))
```

Comment:
There are some operations in the PR that could be accelerated using multiprocessing such as:

        with Pool() as p:
            masks = p.map(
                    self.filter_color,
                    [
                        (transformed_mask, np.array([220, 220, 220]), np.array([240, 240, 240])),
                        (transformed_mask, np.array([180, 180, 180]), np.array([200, 200, 200])),
                        (transformed_mask, np.array([160, 160, 160]), np.array([180, 180, 180])),
                        (transformed_mask, np.array([240, 240, 240]), np.array([255, 255, 255])),
                    ]
            )
However, I am facing:
`AssertionError: daemonic processes are not allowed to have children`

Reply:
@SiddhantBahuguna 
Threading brings no advantage ?
Else have you tried (py3.8) import concurrent.futures.ProcessPoolExecutor as Pool ?

Reply:
@felixdittrich92 
Thanks for taking time to resolve this :) 
I gave concurrent.futures.ProcessPoolExecutor a try, and the same Assertion Error popped up. May be we require/employ something non-daemonic for the workers process to avoid this.